src/components/CreateAutomationModal.tsx

```diff
@@ -1,19 +1,47 @@
-import { useContext } from "react"
-import { ModalContext } from "~/state/ModalContext"
+import { type MouseEvent, useContext } from "react"
+import { z } from "zod";
+import { ModalContext, type ModalContextType } from "~/state/ModalContext"
 
-const CreateAutomationModal = () => {
+const eventTypeSchema = z.array(z.object({
+  id: z.number(),
+  name: z.enum(['Calendar', 'Mail', 'GitHub', '']),
+  description: z.string()
+}))
+
+type eventTypes = z.infer<typeof eventTypeSchema>
 
-  const { setCreateModalOpened, setRulesetModalOpened } = useContext(ModalContext)
+const CreateAutomationModal = () => {
+  
+  const { setCreateModalOpened, setRulesetModalOpened, setEventTrigger } = useContext(ModalContext)
+  
+  const eventTypes: eventTypes = [
+    { id: 1, name: 'Mail', description: 'Example: When an email was receieved...' },
+    { id: 2, name: 'GitHub', description: 'Example: When a PR has been merged to a repository...' },
+    { id: 3, name: 'Calendar', description: 'Example: When a meeting is canceled...' },
+    // event types
+  ];
+  
+  const handleClickedAutomation = (e: MouseEvent<HTMLDivElement>, eventType: eventTypes[0]["name"]) => {
```

Comment:
nit: instead of having to index into `eventTypes` here, it'd be slightly cleaner to decouple the definition of an event type from being in an array

Reply:
Ahhh I didn't think of that at all. I actually like that approach a lot more tbh